### DISCUSSION TITLE
How to use Babel preset with Expo

### DISCUSSION THREAD

**Discussion by songpola** [microsoft/rnx-kit#2753] [Q&A] [answered]
I want to use `@rnx-kit/babel-preset-metro-react-native` with Expo.
But Expo already has existing `babel-preset-expo`.
Both have stated that they're a drop-in replacement for `metro-react-native-babel-preset`.

So, should I use only one:
`{ "presets": ["@rnx-kit/babel-preset-metro-react-native"] }`,

or should I use both at the same time?
`{ "presets": ["babel-preset-expo", "@rnx-kit/babel-preset-metro-react-native"] }`,

(Also in which order?)
`{ "presets": ["@rnx-kit/babel-preset-metro-react-native", "babel-preset-expo"] }`,

**Comment by tido64** [comment, marked as answer]
Had a quick look at [`babel-preset-expo`](https://github.com/expo/expo/blob/master/packages/babel-preset-expo/index.js), it looks like it imports `metro-react-native-babel-preset` and adds more things on top of it. `@rnx-kit/babel-preset-metro-react-native` does the same thing. If you use both, we cannot guarantee that the same version of `metro-react-native-babel-preset` is used for both. Since you're using Expo, I would stick to only using `babel-preset-expo`, and only add what you need from `@rnx-kit/babel-preset-metro-react-native`. If you know what you need from it, we can at least make it easier for you to import those things.

**Reply by songpola** [reply]
What `@rnx-kit/babel-preset-metro-react-native` has and `babel-preset-expo` does not?
I mean. what would I need from `@rnx-kit/babel-preset-metro-react-native` if it does the same thing as `babel-preset-expo` ?

**Reply by tido64** [reply]
To be honest, I haven't looked very closely so I don't know. I assumed that since you wanted to add `@rnx-kit/babel-preset-metro-react-native` that you found something useful in there. 😅

**Reply by songpola** [reply]
Sorry for very late reply. I finally figured out why I need to use `@rnx-kit/babel-preset-metro-react-native` in the first place.
It because I want to use `@rnx-kit/metro-config` and it shows that I need that Babel preset. ([Docs](https://microsoft.github.io/rnx-kit/docs/tools/metro-config))
So, I wonder what does the Babel preset really do?
Can I use `@rnx-kit/metro-config` without it? (Using Expo's instead.)

**Reply by tido64** [reply]
Right, that looks wrong. You should be able to use `@rnx-kit/metro-config` without the Babel preset. If it doesn't work, let us know so we can fix it.

Edit: Updated docs: https://github.com/microsoft/rnx-kit/pull/2770
